reporter.js

```diff
@@ -3,8 +3,53 @@ var throughObj = require('through2').obj
 
 const PLUGIN_NAME = 'gulp-pug-linter'
 
-module.exports = function (hasToFail) {
+var defaultReporter = function (errors) {
+  if (errors.length) {
+    var allErrors = errors.map(function (error) {
+      return error.message
+    }).join('\n\n')
+
+    gutil.log(allErrors)
+  }
+}
+
+var failReporter = function (errors) {
+  if (errors.length) {
+    var allErrors = errors.map(function (error) {
+      return error.message
+    }).join('\n\n')
+
+    this.emit('error', new gutil.PluginError(PLUGIN_NAME, allErrors))
+  }
+}
+
+var loadReporter = function (type) {
```

Comment:
Please annotate this reporter with something like this:

```js
/**
 * @name loadReporter
 * @description Returns an error reporter
 * @param {*} [type] 'fail' string to load a reporter that fails on error,
 *                   any function or string to load a custom module reporter,
 *                   or empty to load the default reporter.
 * @returns {Function} Error reporter that accepts an array of errors
 */
```

Please note that in this case, the `type` parameter is [optional](http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values).